mage_hashtarget.go

```diff
@@ -0,0 +1,224 @@
+// +build mage
```

Comment:
can you move all these go files in a dedicated dir? I suggest `./go/internal/mage`

Reply:
btw, you can keep the magefile.go at the root, its dependencies should not pollute the root dir

Reply:
I tried at first, will try again